### ISSUE TITLE
More Piles 1.0.6 - Placing a rope ladder pile crashes

### ISSUE THREAD

**Issue by GPlayah** [Craluminum2413/Craluminum-Mods#3]
**Game Version:** v1.15.9 (Stable)
**Mod Version:** 1.0.6

**Description:**
Game crashes when placing a rope ladder pile on the ground.

**How to reproduce:**
1. Hold rope ladders
2. Aim at the top surface of an empty block
3. Sneak, and place ladder

**Expected behavior:**
Place a rope ladder pile.

**Log:**
```
Running on 64 bit Windows with 16 GB RAM 
Version: v1.15.9 (Stable)11/18/2021 8:53:48 PM: Critical error occurred
System.NullReferenceException: Object reference not set to an instance of an object.
   at Vintagestory.GameContent.BlockEntityGroundStorage.get_Item(String textureCode)
   at Vintagestory.Client.NoObf.ShapeTesselator.TesselateShapeElement(Int32 indexForLogging, MeshData meshdata, ShapeElement element, TesselationMetaData meta)
   at Vintagestory.Client.NoObf.ShapeTesselator.TesselateShapeElements(MeshData meshdata, ShapeElement[] elements, TesselationMetaData meta)
   at Vintagestory.Client.NoObf.ShapeTesselator.TesselateShape(Shape shapeBase, MeshData& modeldata, Vec3f wholeMeshRotation, Vec3f wholeMeshOffset, Single wholeMeshScale, TesselationMetaData meta)
   at Vintagestory.Client.NoObf.ShapeTesselator.TesselateShape(String typeForLogging, Shape shapeBase, MeshData& modeldata, ITexPositionSource texSource, Vec3f wholeMeshRotation, Int32 generalGlowLevel, Byte climateColorMapId, Byte seasonColorMapId, Nullable`1 quantityElements, String[] selectiveElements)
   at Vintagestory.GameContent.BlockEntityGroundStorage.getOrCreateMesh(ItemSlot itemSlot, Vec3f offset)
   at Vintagestory.GameContent.BlockEntityGroundStorage.updateMeshes()
   at Vintagestory.GameContent.BlockEntityGroundStorage.OnPlayerInteract(IPlayer player, BlockSelection bs)
   at Vintagestory.GameContent.BlockGroundStorage.CreateStorage(IWorldAccessor world, BlockSelection blockSel, IPlayer player)
   at Vintagestory.GameContent.CollectibleBehaviorGroundStorable.Interact(ItemSlot itemslot, EntityAgent byEntity, BlockSelection blockSel, EntitySelection entitySel, Boolean firstEvent, EnumHandHandling& handHandling, EnumHandling& handling)
   at Vintagestory.API.Common.CollectibleObject.<>c__DisplayClass91_0.<OnHeldInteractStart>b__0(CollectibleBehavior bh, EnumHandling& hd)
   at Vintagestory.API.Common.CollectibleObject.WalkBehaviors(CollectibleBehaviorDelegate onBehavior, Action defaultAction)
   at Vintagestory.API.Common.CollectibleObject.OnHeldInteractStart(ItemSlot slot, EntityAgent byEntity, BlockSelection blockSel, EntitySelection entitySel, Boolean firstEvent, EnumHandHandling& handling)
   at Vintagestory.Client.NoObf.SystemMouseInWorldInteractions.TryBeginUseActiveSlotItem(BlockSelection blockSel, EntitySelection entitySel, EnumHandInteract useType, EnumHandHandling& handling)
   at Vintagestory.Client.NoObf.SystemMouseInWorldInteractions.TryBeginUseActiveSlotItem(BlockSelection blockSel, EntitySelection entitySel)
   at Vintagestory.Client.NoObf.SystemMouseInWorldInteractions.HandleMouseInteractionsBlockSelected(Single dt)
   at Vintagestory.Client.NoObf.SystemMouseInWorldInteractions.OnFinalizeFrame(Single dt)
   at Vintagestory.Client.NoObf.ClientEventManager.TriggerRenderStage(EnumRenderStage stage, Single dt)
   at Vintagestory.Client.NoObf.ClientMain.TriggerRenderStage(EnumRenderStage stage, Single dt)
   at Vintagestory.Client.NoObf.ClientMain.RenderToDefaultFramebuffer(Single dt)
   at _rnlVSzJ2AGPu2719D3re8Z7kN2c._TshCCJvNc6laaisiRTK6RNaXWQ1A(Single )
   at _o0lbNJXQO06me4d8j661Kh7y0zO._HmSBpw6fP4XFLEcv29tc0ifOKYvb(Single )
   at _o0lbNJXQO06me4d8j661Kh7y0zO._3U99eoPu0PaDLPVrL8tfzH5UDEJ(Single )
   at Vintagestory.Client.NoObf.ClientPlatformWindows.window_RenderFrame(Object sender, FrameEventArgs e)
   at System.EventHandler`1.Invoke(Object sender, TEventArgs e)
   at OpenTK.GameWindow.RaiseRenderFrame(Double elapsed, Double& timestamp)
   at OpenTK.GameWindow.DispatchRenderFrame()
   at OpenTK.GameWindow.Run(Double updates_per_second, Double frames_per_second)
   at _M48MnHH5BAvHUzcYoEuchRW0acb._e0atQ8VmmvgBK7zb3Yh4iDqOiHY(_UGHNIsOGTJBe1BTRXYmLwHTbmS , String[] )
   at _xNqQDGqRsRSMIeuzt2MHKQHDzSd._e0atQ8VmmvgBK7zb3Yh4iDqOiHY(ThreadStart )
-------------------------------

Event Log entries containing Vintagestory.exe, the latest 3
==================================
{ TimeGenerated = 11/2/2021 8:35:46 PM, Site = , Source = Windows Error Reporting, Message = Fault bucket 1478791943016609661, type 5
Event Name: RADAR_PRE_LEAK_64
Response: Not available
Cab Id: 0

Problem signature:
P1: Vintagestory.exe
P2: 1.15.7.0
P3: 10.0.19042.2.0.0
P4: 
P5: 
P6: 
P7: 
P8: 
P9: 
P10: 

Attached files:
\\?\C:\Users\playa\AppData\Local\Temp\RDR9E67.tmp\empty.txt
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER9E68.tmp.WERInternalMetadata.xml
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER9E78.tmp.xml
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER9E6E.tmp.csv
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER9E9E.tmp.txt

These files may be available here:


Analysis symbol: 
Rechecking for solution: 0
Report Id: 71c91376-d526-46ec-ac24-e54e42453843
Report Status: 268435456
Hashed bucket: 2654803b8137c9295485b96fec36f77d
Cab Guid: 0 }
--------------
{ TimeGenerated = 10/1/2021 1:24:49 PM, Site = , Source = Windows Error Reporting, Message = Fault bucket 1298668709714251717, type 5
Event Name: RADAR_PRE_LEAK_64
Response: Not available
Cab Id: 0

Problem signature:
P1: Vintagestory.exe
P2: 1.15.6.0
P3: 10.0.19042.2.0.0
P4: 
P5: 
P6: 
P7: 
P8: 
P9: 
P10: 

Attached files:
\\?\C:\Users\playa\AppData\Local\Temp\RDR5C4A.tmp\empty.txt
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER5C5B.tmp.WERInternalMetadata.xml
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER5C6B.tmp.xml
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER5C89.tmp.csv
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER5CA9.tmp.txt

These files may be available here:


Analysis symbol: 
Rechecking for solution: 0
Report Id: fba2207a-d671-449f-90d8-4444bcd7c563
Report Status: 268435456
Hashed bucket: 8f39dc2e068300f29205cc4f936ecfc5
Cab Guid: 0 }
--------------
{ TimeGenerated = 9/1/2021 1:18:11 AM, Site = , Source = Windows Error Reporting, Message = Fault bucket 1658733256924091269, type 5
Event Name: RADAR_PRE_LEAK_64
Response: Not available
Cab Id: 0

Problem signature:
P1: Vintagestory.exe
P2: 1.15.5.0
P3: 10.0.19042.2.0.0
P4: 
P5: 
P6: 
P7: 
P8: 
P9: 
P10: 

Attached files:
\\?\C:\Users\playa\AppData\Local\Temp\RDRF5B1.tmp\empty.txt
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WERF5B2.tmp.WERInternalMetadata.xml
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WERF5C3.tmp.xml
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WERF5CE.tmp.csv
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WERF5EE.tmp.txt

These files may be available here:


Analysis symbol: 
Rechecking for solution: 0
Report Id: 14608ab1-82c8-4ea0-997f-c161f24ab518
Report Status: 268435456
Hashed bucket: eb8d9853264150393705011bdc165385
Cab Guid: 0 }
```

**Comment by Craluminum2413** [comment]
Oh, it looks like I forgot to add a separated shape for rope pile, I will add it in 1.0.7

**Comment by Craluminum2413** [comment]
Just fixed and released 1.0.7
